probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -203,6 +208,25 @@ func (s *DNSSnooper) run() {
 	}
 }
 
+// handleDecodeError logs errors up to the maximum allowed count
+func (s *DNSSnooper) handleDecodingError(err error) {
+	// prevent potential memory leak
+	if len(s.decodingErrorCounts) > maxDecodingErrorCardinality {
```

Comment:
If we ever get there we would need to rethink the approach. It's a bit of a hack.